bindings/ros/aditof_roscpp/src/examples/rviz_pointcloud.cpp

```diff
@@ -51,19 +54,29 @@ int main(int argc, char **argv) {
 
     applyNoiseReduction(camera, argc, argv);
 
-    AditofSensorMsg *pclMsg =
-        MessageFactory::create(camera, MessageType::sensor_msgs_PointCloud2);
+    Frame frame;
+    getNewFrame(camera, &frame);
+
+    AditofSensorMsg *msg = MessageFactory::create(
+        camera, &frame, MessageType::sensor_msgs_PointCloud2);
 
-    if (!pclMsg) {
+    if (!msg) {
         ROS_ERROR("pointcloud message creation failed");
+        return -1;
     }
 
     while (ros::ok()) {
-        dynamic_cast<PointCloud2Msg *>(pclMsg)->FrameDataToMsg(camera);
+        getNewFrame(camera, &frame);
+        PointCloud2Msg *pclMsg = dynamic_cast<PointCloud2Msg *>(pclMsg);
+
+        if (!pclMsg) {
+            ROS_ERROR("downcast from AditofSensorMsg to PointCloud2Msg failed");
+            return -1;
+        }
+        dynamic_cast<PointCloud2Msg *>(pclMsg)->FrameDataToMsg(camera, &frame);
```

Comment:
Is this dynamic_cast necessary? You are already down-casting on line 70
```cpp
PointCloud2Msg *pclMsg = dynamic_cast<PointCloud2Msg *>(pclMsg);
```
so `pclMsg` is already of type `PointCloud2Msg`.
Also shouldn't it be `dynamic_cast<PointCloud2Msg *>(msg)` instead of `dynamic_cast<PointCloud2Msg *>(pclMsg)`. Maybe I'm missing something

Reply:
You're right. I made the changes requested in previous PR comments to replace line 76 and then forgot to delete that line. Fixed now, thanks.